src/Compilers/Core/Portable/MetadataReference/MetadataReference.cs

```diff
@@ -232,15 +238,19 @@ public MetadataReference WithProperties(MetadataReferenceProperties properties)
             var peStream = FileUtilities.OpenFileStream(path);
 
             // prefetch image, close stream to avoid locking it:
+#pragma warning disable CA2000 // Dispose objects before losing scope - MetadataImageReference has dispose ownership
```

Comment:
>#pragma warning disable  [](start = 0, length = 24)

Can we have just one suppression for the entire method instead of two? These #pragmas add a lot of clutter.

Reply:
:memo: I found the separation tremendously helpful during review. I would not have been able to complete the review with another approach.